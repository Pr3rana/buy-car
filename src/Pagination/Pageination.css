.pagination-container{
    text-align: center;
}
.pagination-container span{
    padding: 12px;
    cursor: pointer;
}
