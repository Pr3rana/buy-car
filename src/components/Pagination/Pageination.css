.pagination-container{
    font-size: 14px;
    text-align: center;
    margin-bottom: 24px;
}
.pagination-container span{
    padding: 12px;
    cursor: pointer;
}
