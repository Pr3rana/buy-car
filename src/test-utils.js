import React from 'react';
import { render } from '@testing-library/react'
import { BrowserRouter as Router, Route, Switch, Link } from 'react-router-dom'; 
import { ListPageContext } from "./helpers/appContext";

export const carData = {
    cars: [{"stockNumber":10000,"manufacturerName":"Tesla","modelName":"Model S","color":"silver","mileage":{"number":151884,"unit":"km"},"fuelType":"Diesel","pictureUrl":"https://auto1-js-task-api--mufasa71.repl.co/images/car.svg"}],
    totalPageCount:100,
    totalCarsCount:1000
}
export const isFetched = false;
const setTotalCarsCount = jest.fn();
const setTotalPageCount = jest.fn();
const setPagenumber = jest.fn();
const setSelectedFilter = jest.fn();
const setSavedCars = jest.fn();
const setAvailableFilters = jest.fn();
const setFilterParams = jest.fn();

const pageContextProvider = {totalCarsCount:1,
    setTotalCarsCount,
    totalPageCount:2,
    setTotalPageCount,
    pagenumber:1,
    setPagenumber,
    selectedFilter:[{}],
    setSelectedFilter,
    savedCars:[],
    setSavedCars,
    availableFilters:{test:["test value"]},
    setAvailableFilters,
    filterParams: [{test:"test filter value"}],
    setFilterParams
}

const AllTheProviders = ({ children }) => {
    return (
     <Router>
          <ListPageContext.Provider value={pageContextProvider}>{children}</ListPageContext.Provider>
     </Router>
    )
}

const customRender = (ui, options) => {
    render(ui, { wrapper: AllTheProviders, ...options });
};

export * from "@testing-library/react";
export { customRender as render };