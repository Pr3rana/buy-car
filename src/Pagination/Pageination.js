import './Pageination.css';
import { useContext, useEffect, useState } from 'react';
import { ListPageContext } from "../helpers/storeContext";

const Pagination = () => {
    const [pageCountLimit, setPageCountLimit] = useState(null);
    const {pagenumber, setPagenumber, totalPageCount} = useContext(ListPageContext);
    
    useEffect(()=>{
        setPageCountLimit(totalPageCount > 10 ? 10 : totalPageCount);
    },[totalPageCount])

    return ( 
        <div className="pagination-container">
            <span key="First" onClick = {()=>setPagenumber(1)}>First</span>
            <span key="Previous" onClick = {()=>setPagenumber(prev=>(prev>1 && prev<=pageCountLimit)?prev-1:prev)}>Previous</span>
            <span key="Page-count">Page {pagenumber} of {pageCountLimit}</span>
            <span key="Next" onClick = {()=>setPagenumber(prev=>(prev>=1 && prev<pageCountLimit)?prev+1:prev)}>Next</span>
            <span key="last" onClick = {()=>setPagenumber(pageCountLimit)}>Last</span>
        </div>
     );
}
 
export default Pagination;
