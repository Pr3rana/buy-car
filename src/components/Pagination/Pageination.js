import './Pageination.css';
import { useContext, useEffect, useState } from 'react';
import { ListPageContext } from "../../helpers/appContext";

const Pagination = () => {
    const [pageCountLimit, setPageCountLimit] = useState(null);
    const {pagenumber, setPagenumber, totalPageCount} = useContext(ListPageContext);
    
    useEffect(()=>{
        setPageCountLimit(totalPageCount > 10 ? 10 : totalPageCount);
    },[totalPageCount])

    return ( 
        <div  data-testid="pagination-container" className="pagination-container">
            <span data-testid="first-page" key="First" onClick = {()=>setPagenumber(1)}>First</span>
            <span data-testid="prev-page" key="Previous" onClick = {()=>setPagenumber(prev=>(prev>1 && prev<=pageCountLimit)?prev-1:prev)}>Previous</span>
            <span data-testid="curr-total-page" key="Page-count">Page {pagenumber} of {pageCountLimit}</span>
            <span data-testid="next-page" key="Next" onClick = {()=>setPagenumber(prev=>(prev>=1 && prev<pageCountLimit)?prev+1:prev)}>Next</span>
            <span data-testid="last-page" key="last" onClick = {()=>setPagenumber(pageCountLimit)}>Last</span>
        </div>
     );
}
 
export default Pagination;
