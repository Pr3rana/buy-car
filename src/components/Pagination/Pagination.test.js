import { render, screen, cleanup } from '../../test-utils.js';
import Pageination from './Pageination';

afterEach(cleanup);
describe("Pageination render",()=>{
    test('Initial rendering', ()=>{
        render(<Pageination />)
        expect(screen.getByTestId('pagination-container')).toMatchSnapshot(`
        <div data-testid="pagination-container" class="pagination-container"><span data-testid="first-page">First</span>
        <span data-testid="prev-page">Previous</span><span data-testid="curr-total-page">Page 1 of 2</span>
        <span data-testid="next-page">Next</span><span data-testid="last-page">Last</span></div>
        `)
    })
})

describe("Pageination Props",()=>{
    test('Testing pagination props', ()=>{
        render(<Pageination />)
        expect(screen.getByText('Page 1 of 2')).toBeInTheDocument()
        expect(screen.getByTestId('first-page')).toHaveTextContent('First')
    })
})
