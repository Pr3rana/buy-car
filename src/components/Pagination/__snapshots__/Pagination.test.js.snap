// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pageination render Initial rendering: 
        <div data-testid="pagination-container" class="pagination-container"><span data-testid="first-page">First</span>
        <span data-testid="prev-page">Previous</span><span data-testid="curr-total-page">Page 1 of 2</span>
        <span data-testid="next-page">Next</span><span data-testid="last-page">Last</span></div>
         1`] = `
<div
  class="pagination-container"
  data-testid="pagination-container"
>
  <span
    data-testid="first-page"
  >
    First
  </span>
  <span
    data-testid="prev-page"
  >
    Previous
  </span>
  <span
    data-testid="curr-total-page"
  >
    Page 
    1
     of 
    2
  </span>
  <span
    data-testid="next-page"
  >
    Next
  </span>
  <span
    data-testid="last-page"
  >
    Last
  </span>
</div>
`;
